### ISSUE TITLE
Problem when deploying an app with dependencies to a runtime package

### ISSUE THREAD

**Issue by freddydk** [microsoft/AL-Go#422]
Originated from Yammer: https://www.yammer.com/dynamicsnavdev/#/Threads/show?threadId=2148170595680256

In the same project I asked a question about moments ago, I'm also experiencing dependency related errors in the "CI/CD" workflow.

I am using the using the microsoft/AL-Go-AppSource template and have configured Environments to publish the latest code to a SaaS environment on push.  If I remove all 3rd party app dependencies, the deployment step works just fine.  

As soon as I define a dependency on another Appsource App (the apps are already installed in the environment), I see the following summarized messages in the Deploy workflow log:

Extensions before:
- Lanham App Library, Version 18.1.5.0, Installed=True
- Lanham AWE OWR Shared Library, Version 21.0.0.1, Installed=True
- Lanham E-Ship and E-Receive, Version 21.1.12.0, Installed=True
- Lanham E-Ship FedEx, Version 20.6.0.9, Installed=True
- Lanham E-Ship UPS, Version 21.4.0.0, Installed=True
- Lanham Outbound Warehouse Request, Version 21.0.1.6, Installed=True
- Lanham Print Queue Library, Version 21.0.0.2, Installed=True

WARNING: Dependency f8750ab4-1d72-455d-b773-5b487a0ac106:Lanham Associates_Lanham E-Ship and E-Receive_20.1.11.3).app not found

WARNING: Dependency 8c70d081-5c33-431c-a585-66ed91a6f257:Lanham Associates_Lanham App Library_18.1.5.0).app not found

WARNING: Dependency 23503827-a4ca-4a57-b616-f2eabb4f4a07:Lanham Associates_Lanham E-Ship FedEx_20.6.0.9).app not found

WARNING: Dependency 8f6fdc64-fde7-41d1-b245-aca18408b645:Lanham Associates_Lanham E-Ship UPS_20.1.11.0).app not found

WARNING: Dependency 9a107513-d044-4008-9dc7-961d3101bdce:Lanham Associates_Lanham AWE OWR Shared Library_21.0.0.0).app not found

WARNING: Dependency ad4cd316-d40c-46ba-ab75-bcd62cea403a:Lanham Associates_Lanham Outbound Warehouse Request_21.0.0.2).app not found

WARNING: Dependency 9545b861-3df4-4720-8389-c0a808166fdc:Lanham Associates_Lanham Print Queue Library_21.0.0.0).app not found

Extensions after:
- Lanham App Library, Version 18.1.5.0, Installed=True
- Lanham AWE OWR Shared Library, Version 21.0.0.1, Installed=True
- Lanham E-Ship and E-Receive, Version 21.1.12.0, Installed=True
- Lanham E-Ship FedEx, Version 20.6.0.9, Installed=True
- Lanham E-Ship UPS, Version 21.4.0.0, Installed=True
- Lanham Outbound Warehouse Request, Version 21.0.1.6, Installed=True
- Lanham Print Queue Library, Version 21.0.0.2, Installed=True
 
Publish-PerTenantExtensionApps Telemetry Correlation Id: fe710c2e-5ded-4072-8645-e14485599cef

Error: Deploying to MYSAASENV failed.  Unable to publish app. Please open the Extension Deployment Status Details page in Business Central to see the detailed error message.

Inside BC, I see the following error:

"Package validation failed due to the following error(s): Error AVS0109: The per-tenant extension (or one of its dependencies) cannot be deployed as it has missing dependencies or the dependencies are conflicting with currently installed apps. Check if these dependencies are installed."

 

I can use VS Code and publish my app to this environment successfully.
I can create an environment locally using localDevEnv.ps1 successfully.
If I update this environment by using the "Create Online Dev. Environment" workflow and set the Reuse Environment flag to Y, the installation is successful.
I cannot get CI/CD to work for automated deployments.
 

I also tried setting "generateDependencyArtifact": true in .AL-Go/settings.json and that gets rid of the WARNING messages mentioned above, but I get a new error of: 

"Publish-PerTenantExtensionApps Telemetry Correlation Id: cf33f20b-2dd8-40b6-ab9a-faf3c8321db5

Error: Deploying to MYSAASENV failed.  AppFile C:\Some\path\Lanham Associates_Lanham App Library_18.1.5.0_Runtime.app is a runtime package. You will have to specify a running container in containerName in order to analyze dependencies between runtime packages"

 **Am I missing a configuration that would cause these errors?  Any ideas would be appreciated.**

**Comment by freddydk** [comment]
It looks like your app has a dependency to these apps:
![image](https://user-images.githubusercontent.com/10775043/221468159-64f7148d-01bb-490d-86ff-0809f57180ba.png)
And not all of these are installed in the tenant.

**Comment by BMI-Mike** [comment]
I can confirm that the three mentioned Versapay apps are and were installed in the SaaS environment at the time I reported the issue.  I only omitted them from the report for brevity (it was a repeat of the same issue found with Lanham apps).

**Comment by BMI-Mike** [comment]
Do you still need access to this repo to investigate?  At current, the use of 3rd party apps is working in local Docker environments for development purposes, but updating our internal SaaS environments has been failing for over 3 weeks.

**Comment by freddydk** [comment]
Hope to have a fix for this next week.

**Comment by BMI-Mike** [comment]
A quick question related to this ticket.  Since the dependent apps listed above are also actually available in AppSource, should I be referencing these apps in a different manner so that a runtime is not needed?  I believe I am doing this correctly, but thought it wise to ask.

**Comment by BMI-Mike** [comment]
Is there an update on this issue?  The deployments have not been working for nearly 2 months.

**Comment by freddydk** [comment]
Sorry, I will make sure that we prioritize open issues when coming back from Directions.

**Comment by freddydk** [comment]
The problem here is in BcContainerHelper.
When publishing apps to an online environment, it should just ignore runtime packages alltogether, instead of trying to sort them and investigate them one-by-one.

The fix will be in the next preview of BcContainerHelper (should be out in 1-2 hours)

You should be able to add a setting to .github\AL-Go-Settings.json like:

`"bcContainerHelperVersion": "preview"
`
To use this version and allow your deploy to work.

**Comment by BMI-Mike** [comment]
I just updated the .github\AL-Go-Settings.json file in one of our projects a few minutes ago.  Then, I ran "Update AL-Go System Files" just to make sure all settings in .github\AL-Go-Settings.json were honored.  Next, I ran the CI/CD workflow manually.

At this time, I'm still receiving the same error message as before that relate to the runtime packages.

The workflow actions are seeing the BcContainerHelperVersion: preview setting from the .github\AL-Go-Settings.json file, but I also see that BcContainerHelper is reporting version 4.0.16 used in the run.  I'm not sure that the workflows are really using the preview version of BcContainerHelper as expected.
